pkg/kv/kvserver/store.go

```diff
@@ -1852,7 +1852,7 @@ func (s *Store) Start(ctx context.Context, stopper *stop.Stopper) error {
 		return err
 	}
 	for fullID, desc := range engRepls.Initialized {
-		rep, err := newReplica(ctx, desc, s, fullID.ReplicaID)
+		rep, err := loadReplica(ctx, s, desc, fullID.ReplicaID)
```

Comment:
TODO: Here is the integration point with the "global" invariant checking and fixing. Instead of `loadReplica`, we can manually construct `loadedReplicaState` from the batch-loaded set of range descs, etc; then `check` invariants, then `newReplica` manually.

Reply:
Or `LoadAndReconcileReplicas` already checks all of the invariants. This makes sense to me, because the invariants are all on data that's on storage, and by having them in the leaf package they get better testing coverage.